packages/snap/src/keyring.ts

```diff
@@ -176,24 +188,81 @@ export class SimpleKeyring implements Keyring {
   // In an asynchronous implementation, the request should be stored in queue
   // of pending requests to be approved or rejected by the user.
   async submitRequest(request: KeyringRequest): Promise<SubmitRequestResponse> {
-    const { method, params = '' } = request.request as JsonRpcRequest;
-    const signature = this.#handleSigningRequest(method, params);
-    return {
-      pending: false,
-      result: signature,
-    };
+    if (this.#useSynchronousApprovals) {
+      return this.#handleSynchronousSubmitRequest(request);
+    }
+    return this.#handleAsyncSubmitRequest(request);
   }
 
   async approveRequest(_id: string): Promise<void> {
```

Comment:
This variable has the `_` prefix because it wasn't used, should be fine to rename it to `id`.